src/Opdex.Platform.WebApi/IndexerBackgroundService.cs

```diff
@@ -45,21 +48,12 @@ protected override async Task ExecuteAsync(CancellationToken cancellationToken)
         {
             while (!cancellationToken.IsCancellationRequested)
             {
-                if (started)
-                {
-                    // delay 30 seconds when indexing services are unavailable (db flag off)
-                    // delay 8 to 12 seconds when indexing is available
-                    var seconds = unavailable ? 30 : new Random().Next(8, 13);
-
-                    await Task.Delay(TimeSpan.FromSeconds(seconds), cancellationToken);
-                }
-
                 try
                 {
                     var indexLock = await mediator.Send(new RetrieveIndexerLockQuery(), cancellationToken);
-                    if (!indexLock.Available)
+                    if (!indexLock.Available || !_indexerOptions.CurrentValue.Enabled)
```

Comment:
Do we need `CurrentValue`?

Reply:
Yeah, this is because we're using `IOptionsMonitor` which updates with config changes. Without it, we'd load config once and it would never change, even if we updated the values in the config source